ukis_pysat/data.py

```diff
@@ -74,48 +81,71 @@ def __init__(self, datahub, catalog=None):
     def __enter__(self):
         return self
 
-    def init_catalog(self, catalog=None):
-        """ Initializes PySTAC Catalog
+    def _init_catalog(self):
+        """Initializes an empty Spatio Temporal Asset Catalog."""
+        return pystac.catalog.Catalog(
+            id=str(uuid.uuid4()),
+            description=f"Creation Date: {datetime.datetime.now()}, Datahub: {self.src.value}",
+            catalog_type=pystac.catalog.CatalogType.SELF_CONTAINED,
+        )
 
-        :param catalog: PySTAC Catalog or Collection if already exists
+    def add_items_from_directory(self, item_dir, item_substr=None):
+        """Adds metadata items from a directory to a STAC source.
+
+        :param item_dir: Path to directory that holds the metadata items (String).
+        :param item_substr: Optional substring patterns to identify metadata items in directory (List of String).
         """
-        if isinstance(catalog, (pystac.catalog.Catalog, pystac.collection.Collection)):
-            self.catalog = catalog
-        elif isinstance(catalog, (str, Path)):
-            href = Path(catalog).resolve().as_uri()
-            self.catalog = pystac.catalog.Catalog.from_file(href)
-        else:
-            self.catalog = pystac.catalog.Catalog(
-                id=str(uuid.uuid4()),
-                description=f"Creation Date: {datetime.datetime.now()}, Datahub: {self.src.value}",
-                catalog_type=pystac.catalog.CatalogType.SELF_CONTAINED,
+        if self.src == Datahub.STAC:
+            # get all json files in item_dir that match item_substr
+            if item_substr is None:
+                item_substr = [""]
+            item_files = sorted(

```

Comment:
I find this tough to read. If you make `item_dir` a Path and `item_substr` defaults to `'*'` you could simply do:
`item_files = sorted(item_dir.rglob(f"{item_substr}.json"))`